chainstate/src/detail/chainstateref/mod.rs

```diff
@@ -743,6 +743,15 @@ impl<'a, S: BlockchainStorageRead, V: TransactionVerificationStrategy> Chainstat
     }
 
     fn check_tokens_txs(&self, block: &Block) -> Result<(), CheckBlockTransactionsError> {
+        let prev_block_id = block.prev_block_id();
+        let current_height = self
+            .get_gen_block_index(&prev_block_id)?
+            .ok_or(CheckBlockTransactionsError::PropertyQueryError(
+                PropertyQueryError::BlockIndexNotFound(prev_block_id),
```

Comment:
Better use the error PrevBlockIndexNotFound. Also, this is not a non-chalant error... this is an invariant broken. 

Why not use the function that was made for this anyway? `get_previous_block_index_for_check_block()`

Reply:
> Better use the error PrevBlockIndexNotFound. 

Agree, done.

> Why not use the function that was made for this anyway?

It returns incompatible Error type. It doesn't do much so I think it's ok to leave it like this.